components/server/src/user/user-service.ts

```diff
@@ -210,10 +210,31 @@ export class UserService {
      * Identifies the team or user to which a workspace instance's running time should be attributed to
      * (e.g. for usage analytics or billing purposes).
      *
+     * A. Billing-based attribution: If payments are enabled, we attribute all the user's usage to:
+     *   - An explicitly selected billing account (e.g. a team with usage-based billing enabled)
+     *   - Or, we default to the user for all usage (e.g. free tier or individual billing, regardless of project/team)
+     *
+     * B. Project-based attribution: If payments are not enabled (e.g. Self-Hosted), we attribute:
+     *   - To the owner of the project (user or team), if the workspace is linked to a project
+     *   - To the user, iff the workspace is not linked to a project
+     *
      * @param user
      * @param projectId
      */
     async getWorkspaceUsageAttributionId(user: User, projectId?: string): Promise<string | undefined> {
+        // A. Billing-based attribution
+        if (this.config.enablePayment) {
```

Comment:
Thx a lot for this PR, @jankeromnes ! :pray: 

However, this is sth I wanted to avoid: having the the attribution logic coupled with user/workspace services. :slightly_smiling_face: 

I'd rather move the (re-)attribution part over to the usage service. Sorry if I wasn't clear there - got a :exploding_head: after all those meetings this week.. :grimacing:

Reply:
Happy to sync about this early next week to discuss and align. :calendar:

Reply:
If you don't like the `if (this.config.enablePayment)` check, we could also make this a `if (user.usageAttributionId)` check (i.e. only "force" the attribution if the user "wants" a specific attribution ID for all their instances)

Reply:
As discussed, this can still be adjusted in follow-up PRs, for example when we implement the entitlement service. 👍